### PR TITLE
feat:Search and Filter in accounts

### PR DESCRIPTION
**Pull request by dilpreet2028**
Demo: 

 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by dilpreet96** [comment]
@therajanmaurya please review it

File: app/src/main/java/org/mifos/selfserviceapp/ui/adapters/CheckBoxAdapter.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        @BindView(R.id.cb_status_select)
+        AppCompatCheckBox cbStatusSelect;
+
+        @BindView(R.id.tv_status)
+        TextView tvStatus;
+
+        public ViewHolder(View itemView) {
+            super(itemView);
+            ButterKnife.bind(this, itemView);
+            itemView.setOnClickListener(new View.OnClickListener() {
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/adapters/CheckBoxAdapter.java]: @dilpreet96 we are using ButterKnife Please use annotation

File: app/src/main/java/org/mifos/selfserviceapp/ui/fragments/ClientAccountsFragment.java [hunk 2]
```diff
@@ -182,22 +195,196 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
...
+                break;
             case R.id.menu_add_loan:
                 ((BaseActivity) getActivity()).replaceFragment(LoanApplicationFragment.
                         newInstance(LoanState.CREATE), true, R.id.container);
                 break;
         }
         return true;
     }
+
+    private void initSearchSavings(Menu menu) {
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/fragments/ClientAccountsFragment.java]: @dilpreet96 you can make the single generic type of method, By just passing 2 -3 parameter and if statements and can use anywhere. Please do

File: app/src/main/java/org/mifos/selfserviceapp/ui/fragments/ClientAccountsFragment.java [hunk 3]
```diff
@@ -182,22 +195,196 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
...
+            @Override
+            public boolean onQueryTextChange(String newText) {
+                ((AccountsFragment) getChildFragmentManager().findFragmentByTag(
+                        getFragmentTag(2))).searchSharesAccount(newText);
+                return false;
+            }
+        });
+    }
+
+    private void showSavingsFilterDialog() {
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/fragments/ClientAccountsFragment.java]: Same as above

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/selfserviceapp/ui/fragments/ClientAccountsFragment.java (hunk 2 above)
Comment (dilpreet2028) [reply app/src/main/java/org/mifos/selfserviceapp/ui/fragments/ClientAccountsFragment.java]: On it :+1:

File: app/src/main/java/org/mifos/selfserviceapp/ui/fragments/ClientAccountsFragment.java (hunk 3 above)
Comment (dilpreet2028) [reply app/src/main/java/org/mifos/selfserviceapp/ui/fragments/ClientAccountsFragment.java]: On it :+1:

**Comment by dilpreet2028** [comment]
@therajanmaurya changes made

**Review by therajanmaurya — APPROVED** [review]

**Pull request merged**